arch/arm/core/aarch32/fatal.c

```diff
@@ -18,24 +18,25 @@ LOG_MODULE_DECLARE(os);
 
 static void esf_dump(const z_arch_esf_t *esf)
 {
+	const struct __esf *hw_esf = esf->exception_frame;
 	LOG_ERR("r0/a1:  0x%08x  r1/a2:  0x%08x  r2/a3:  0x%08x",
-		esf->basic.a1, esf->basic.a2, esf->basic.a3);
+		hw_esf->basic.a1, hw_esf->basic.a2, hw_esf->basic.a3);
 	LOG_ERR("r3/a4:  0x%08x r12/ip:  0x%08x r14/lr:  0x%08x",
-		esf->basic.a4, esf->basic.ip, esf->basic.lr);
-	LOG_ERR(" xpsr:  0x%08x", esf->basic.xpsr);
+		hw_esf->basic.a4, hw_esf->basic.ip, hw_esf->basic.lr);
+	LOG_ERR(" xpsr:  0x%08x", hw_esf->basic.xpsr);
```

Comment:
Since we now have these extra registers available we should dump them here along with the ones we already print out.